File: backend/SNS.Application/Jobs/Jobs/Contracts/JobSnapshotDto.cs
using SNS.Domain.Jobs.Enums;

namespace SNS.Application.Jobs.Jobs.Contracts;

public sealed record JobSnapshotDto(
    Guid Id,
    string Title,
    string CompanyName,
    string Location,
    JobType Type,
    decimal MinSalary,
    decimal MaxSalary,
    string CurrencyCode,
    SalaryType SalaryType
);

File: SNS.Web/src/app/jobs/jobs/contracts/job-snapshot.dto.ts
import { JobType } from '../../enums/job-type.enum';
import { SalaryType } from '../../enums/salary-type.enum';

export interface JobSnapshotDto {
    id: string;
    title: string;
    companyName: string;
    location: string;
    type: JobType;
    minSalary: number;
    maxSalary: number;
    currencyCode: string;
    salaryType: SalaryType;
}
